sadx-dc-lighting/lantern.cpp

```diff
@@ -438,38 +438,37 @@ bool LanternInstance::load_palette(const std::string& path)
 
 	file.close();
 
-	if (color_data.size() > palette_pairs_.size())
+	unsigned int maxlength = palette_index_count * palette_index_length;
+	if (color_data.size() > maxlength)
 	{
-		PrintDebug("[lantern] WARNING: Palette size %d exceeds standard maximum.\n", color_data.size());
+		PrintDebug("[lantern] WARNING: Palette size %d exceeds standard maximum of %d.\n", color_data.size(), maxlength);
```

Comment:
I know the original format string already had `%d`, but if it's supported by the formatter implemented in SADX, `%u` should probably be used instead, since these values are unsigned.